restcomm/restcomm.monitoring.service/src/main/java/org/restcomm/connect/monitoringservice/MonitoringService.java

```diff
@@ -113,6 +128,15 @@ public MonitoringService(final DaoManager daoManager) {
         maxConcurrentCalls = new AtomicInteger(0);
         maxConcurrentIncomingCalls = new AtomicInteger(0);
         maxConcurrentOutgoingCalls = new AtomicInteger(0);
+
+        mgcpEndpointsBridge = new AtomicInteger(0);
+        mgcpEndpointsIvr = new AtomicInteger(0);
+        mgcpEndpointsConference = new AtomicInteger(0);
+        mgcpEndpointsPacketRelay = new AtomicInteger(0);
+
+        mgcpEndpointMap = new ConcurrentHashMap<String, String>();
```

Comment:
seems we only need the size of these maps to return as part of onStatistics logic... Do we really need to save all this info into maps? it seems a potential place for unnecesary leaks, when the size of these maps can be already calculated by using other counters already present... I would try to reduce state to simple counters, that maybe be wrong but do not risk to introduce more leaks...

Reply:
@jaimecasero I am thinking that in the future we can provide MGCP resource details by call, similar to the live call details, so in that case the Map is needed.
Agree with your point but I don't see a risk of leaks here.